sklearn/metrics/pairwise.py

```diff
@@ -1059,13 +1064,14 @@ def _parallel_pairwise(X, Y, func, n_jobs, **kwds):
     if effective_n_jobs(n_jobs) == 1:
         return func(X, Y, **kwds)
 
-    # TODO: in some cases, backend='threading' may be appropriate
-    fd = delayed(func)
-    ret = Parallel(n_jobs=n_jobs, verbose=0)(
-        fd(X, Y[s], **kwds)
+    # enforce a threading backend to prevent data communication overhead
+    fd = delayed(_dist_wrapper)
+    ret = np.empty((X.shape[0], Y.shape[0]), dtype=X.dtype, order='F')
+    Parallel(backend="threading", n_jobs=n_jobs)(
```

Comment:
we should probably not enforce backend like this be instead use the `prefer="threads"` hinting mechanism. That is using the `**_joblib_parallel_args(prefer='threads')` trick to keep backward compat with joblib 0.11.

Reply:
I was wondering about that, but say when you are using Dask, do we really want it to be able to change the backend here? I can't think of a use case where we want to use process-based backend for Euclidean distances which probably is the most frequent case.

Other metrics might require more fine-grained treatment ...